image.h

```diff
@@ -0,0 +1,20 @@
+#include <cstdint>
+#include <memory>
+
+class Image {
+public:
+    Image(uint32_t rows, uint32_t columns);
+    Image(const Image& image); // Copy constructor

```

Comment:
Can this `Image` class be modified using the `=` operator? This operator is normally generated by default to do a memberwise copy. Maybe having a `unique_ptr` prevents the copying. Or you can explicitly prevent the default `=` operator using `Image& operator=(const Image&) = delete;`

Reply:
agree . changes done now